x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/jdbc/TypeConverter.java

```diff
@@ -94,15 +115,34 @@ static Timestamp convertTimestamp(Long millis, Calendar cal) {
             c.setTimeInMillis(initial);
         }
     }
+    
+    static long convertFromCalendarToUTC(long value, Calendar cal) {
+        if (cal == null) {
+            return value;
+        }
+        Calendar c = (Calendar) cal.clone();
+        c.setTimeInMillis(value);
+
+        ZonedDateTime convertedDateTime = ZonedDateTime
```

Comment:
There should be a way to do the conversion without doing the raw math in there (not to mention the whole /1000, * 1000 losses approximation).

Why not use the `ZoneId` from the calendar to compute the offset:

```
ZonedDateTime zdt = Instant.ofEpochMillis(value).atZone(cal.getTimeZone().toZoneId());
```
or potentially use `getOffset()` instead of `getRawOffset()`

Reply:
Changed the way the conversion is done